spring-boot-actuator/src/main/java/org/springframework/boot/actuate/autoconfigure/HealthIndicatorAutoConfiguration.java

```diff
@@ -276,4 +280,41 @@ public DiskSpaceHealthIndicatorProperties diskSpaceHealthIndicatorProperties() {
 
 	}
 
+	@Configuration
+	@ConditionalOnBean(JavaMailSenderImpl.class)
+	@ConditionalOnProperty(prefix = "management.health.mail", name = "enabled", matchIfMissing = true)
```

Comment:
You need to add a meta-data entry for this key in `META-INF/additional-spring-configuration-metadata.json`. Look at the others for an example.